src/audio/copier/copier_dai.c

```diff
@@ -286,13 +286,26 @@ int copier_dai_create(struct comp_dev *dev, struct copier_data *cd,
 		if (ret)
 			return ret;
 		break;
+	case ipc4_alh_uaol_stream_link_output_class:
+	case ipc4_alh_uaol_stream_link_input_class:
+		dai.type = SOF_DAI_INTEL_UAOL;
+		dai.is_config_blob = true;
+		type = ipc4_gtw_alh;
+		ret = ipc4_find_dma_config(&dai, (uint8_t *)cd->gtw_cfg,
+					   copier->gtw_cfg.config_length * 4);
+		if (ret != IPC4_SUCCESS) {
+			comp_err(dev, "No uaol dma_config found in blob!");
+			return -EINVAL;
+		}
+		dai.out_fmt = &copier->out_fmt;
+		break;
 	case ipc4_dmic_link_input_class:
 		dai.type = SOF_DAI_INTEL_DMIC;
 		dai.is_config_blob = true;
 		type = ipc4_gtw_dmic;
 		ret = ipc4_find_dma_config(&dai, (uint8_t *)cd->gtw_cfg,
 					   copier->gtw_cfg.config_length * 4);
-		if (ret != 0) {
+		if (ret != IPC4_SUCCESS) {
```

Comment:
unrelated change -> different patch.

Reply:
Removed this unrelated change from this patch.